### PR TITLE
Mod/requirements readme update

### PR DESCRIPTION
**Pull request by justinle**
Two known security vulnerabilities fixed by updating Django (1.11.14 -> 1.11.15) and drf-tracking libraries.

Speedup and memory optimization fixes from updating bidict library (0.13.1->0.17.2).

autoenv library no longer supported as of Python 3, third-party Virtualenv library was made redundant as of Python3.3 with native support of virtual environments in its standard library using with `venv`
Replaced autoenv and virtualenv with autoenv's successor `direnv` and Python's `venv` and updated steps in readme.

Reorganized readme to streamline installation and setup, include `direnv` & `venv`, and to added more information as emphasize and clarify certain details.

### CODE REVIEW

File: requirements/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,5 @@
 appdirs==1.4.3
-autoenv==1.0.0
-bidict==0.13.1
+bidict==0.17.2
```
Comment (tony-sappe) [inline requirements/requirements.txt]: Is `bidict` used in the codebase or a dependency for another library?

File: README.md [hunk 2]
```diff
@@ -1,175 +1,302 @@
...
+This is the API that will drive the new USAspending.gov. It is currently under active development and not stable. This API is built on Python 3.5+, Django, and Django Rest Framework.
+
+## Built With
+**[PostgreSQL](https://www.postgresql.org/)** is an object-relational database management system (ORDBMS) with an emphasis on extensibility and standards-compliance.
+
+The API’s backend components currently run on **[Python 3.5](https://www.python.org/downloads/)** or higher. These instructions will walk you through the process of installing Python and creating a Python-based virtual environment to house the API. A virtual environment will isolate the project and its libraries from those running on your local system and prevent potential conflicts.
+
+## Prerequisites
+**Assumptions**:
+* Ability software on your local machine
```
Comment (tony-sappe) [inline README.md]: Perhaps `Ability to download, install, and run software on your local machine`

File: README.md [hunk 3]
```diff
@@ -1,175 +1,302 @@
...
+```
 
-        $ cd usaspending-api
+  
 
+Install caching dependencies. (This is optional, and should only be done if you have installed Memcached and require AWS caching for development)
 
-Install the dependencies.
+```
+$ pip install -r requirements/caching_requirements.txt
```
Comment (tony-sappe) [inline README.md]: include "(python-venv-3.5.0)" or remove from previous pip install instruction for consistency

File: README.md [hunk 4]
```diff
@@ -1,175 +1,302 @@
...
 
-Next, configure your local settings. The `settings.py` file will check for an environment variable named `DATABASE_URL`. You can export this variable on the command line, or use a library like [autoenv](https://github.com/kennethreitz/autoenv) to load your environment variables when you `cd` into that directory.
+## Set your virtual environment export commands
 
-Exporting your `DATABASE_URL` environment variable should look something like this:
+Export your `DATABASE_URL` environment variable by appending the following to your `.envrc` file
 
-        $ export DATABASE_URL='postgres://USER:PASSWORD@HOST:PORT/DATABASENAME'
+```
+$ export DATABASE_URL='postgres://USER:PASSWORD@HOST:PORT/DATABASENAME'
```
Comment (tony-sappe) [inline README.md]: This doesn't work for Windows users does it? Either all instructions should have the necessary details for the three envs or we drop all Windows steps and just provide MacOS/Linux instructors.

I don't know if the Ubuntu integration with Windows 10 might work as a middle ground.

File: README.md [hunk 5]
```diff
@@ -1,175 +1,302 @@
-# USASpending API
 
+<p align="center">
+        <img src="https://avatars1.githubusercontent.com/u/8397762?s=200&v=4" />
+</p>
```
Comment (tony-sappe) [inline README.md]: I think there needs to be a blank line after the </p>

File: README.md [hunk 6]
```diff
@@ -1,175 +1,302 @@
...
-* `load_reference_data` - Loads all reference data (used on a fresh or flushed db)
-* `update_location_usage_flags` - Updates the `place_of_performance_flag` and `recipient_flag` on Location objects based upon their foreign key usages
-* `load_submission` - Loads a submission from the broker for the currently set environment
-* `loadtas` - Loads tas account information from a file
-* `loadagencies` - Loads agency information from a file
-* `loadcfda` - Loads CFDA information from a file
-* `loadcontracts` - Loads USASpending contract information from a file
-* `generate_test_endpoint_responses` - Generates endpoint test responses for testing
+  
+*  `load_reference_data` - Loads all reference data (used on a fresh or flushed db)
```
Comment (tony-sappe) [inline README.md]: We have a number of management commands. I don't know if it makes sense to draw attention to any of them in this README.

File: README.md [hunk 7]
```diff
@@ -1,175 +1,302 @@
...
 ## Running the Test Suite
 
 To run the test suite and confirm that everything is working as expected, make sure you're in the top-level project directory and that your `DATABASE_URL` environment variable is set, and
-run the following from the command line:
 
-
-        py.test
+run the following from the command line:
+```
+$ py.test
```
Comment (tony-sappe) [inline README.md]: `pytest --cov=usaspending_api`

File: requirements/requirements.txt (hunk 1 above)
Comment (justinle) [reply requirements/requirements.txt]: I believe it was used in the past but currently is instructed to be used if needed on `line 349` of  `settings.py`

# TERSE_TO_LONG UNUSED.  IF NEEDED, change LONG_TO_TERSE to a bidict or invert the dict.

Comment (tony-sappe) [reply requirements/requirements.txt]: If it isn't being used I recommend the package with original version (bidict==0.13.1) be moved to `legacy_requirements.txt` for now.

File: README.md [hunk 8]
```diff
@@ -1,173 +1,67 @@
...
-## ETL Setup
-
-The Django server has extract, transform, and load (ETL) processes built in as custom management commands. To use them, follow these setup steps:
-
-  1. Configure the `DATA_BROKER_DATABASE_URL` environment variable. This is the same process as setting up the API backend. Exporting `DATA_BROKER_DATABASE_URL` should look like this:
-
-  ```
-  $ export DATA_BROKER_DATABASE_URL='postgres://USER:PASSWORD@HOST:PORT/DATABASENAME'
-  ```
+Available API endpoints may be found at `http://localhost:8000/docs/endpoints`
```
Comment (kwhickey) [inline README.md]: @justinle In addition to pointing to where a developer could find API docs on their local env, can you also provide the link to the latest published production API docs. That would be:

https://api.usaspending.gov/

_(from there, they should be able to click the "Endpoints" link on the left nav to get further into the api docs)_

File: README.md [hunk 9]
```diff
@@ -1,173 +1,67 @@
-# USASpending API
+<p align="center"><img src="https://www.usaspending.gov/img/logo@2x.png" alt="USASpending API"></p>
```
[the remark this answers is not in the document]
Comment (justinle) [reply README.md]: changed all occurrences of `USASpending` to `USAspending`

File: README.md (hunk 8 above)
Comment (justinle) [reply README.md]: The main endpoints link now points to the main API site, the local environment is also noted under the main link.

**Comment by justinle** [comment]
Revised the readme in response to code review comments

File: requirements/requirements.txt (hunk 1 above)
Comment (justinle) [reply requirements/requirements.txt]: Just saw this, `bidict` has now been moved to `requirements-legacy.txt`

File: README.md (hunk 6 above)
Comment (justinle) [reply README.md]: resolved in new readme

**Pull request merged**